addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -1980,9 +2018,13 @@ const SnippetOptionWidget = Widget.extend({
      * @param {Object} params
      * @returns {Promise|undefined}
      */
-    selectAttribute: function (previewMode, widgetValue, params) {
-        const value = this._selectAttributeHelper(widgetValue, params);
-        this.$target[0].setAttribute(params.attributeName, value);
+    selectAttribute: async function (previewMode, widgetValue, params) {
+        const value = await this._selectAttributeHelper(widgetValue, params);
+        if (previewMode) {
+            this.$target.attr(params.attributeName, value);
+        } else {
+            await this.editorHelpers.setAttribute(this.wysiwyg.editor, this.$target[0], params.attributeName, value);
+        }
```

Comment:
It feels right to me: it's explicit about where the changes are applied. In one case you just want the changes to apply to the dom, in other you want to "commit" them to the editor.
Now that's of course a question of personal preference. But I think the more explicit the better even if that means slightly longer code. Or did you mean something else I didn't understand properly?

Reply:
It's really not a question of personal preference. It's already huge to expect that an external dev has to learn a dedicated Odoo API to modify the DOM, we really should not expect them to learn the JavaScript API + the Odoo API for basic operations such as "I just want to set an attribute on this".

Reply:
Well I respectfully disagree. We're talking about two different operations here. What you call "I just want to set an attribute on this" is actually "I just want to set an attribute on this but without saving it".
To be clear I'm not totally closed off to the idea of doing it your way, but I have to disagree when you imply it's an objective truth.

Reply:
> What you call "I just want to set an attribute on this" is actually "I just want to set an attribute on this but without saving it".

The part "but without saving it" should not mean anything for designers, especially the ones outside Odoo who cannot reach out to you. Taking care of the DOM through JS is complex enough for someone who just want to design a snippet, adding the complexity of going through Odoo's functions is something we could agree on... but how would explain to designers that they basically have to code their method twice and that Odoo does not take care of it for them?

We can disagree of course, I was not saying I have the absolute truth :shrug: I just strongly believe that I am right, with the experience of having seen what kind of crappy code we already can find in those methods up to 14.0. So if you ask people to code what they want to do twice... you just increase the chance of the function not working (in previewmode and or not in previewmode).

Reply:
This discussion is sterile for the simple reason that AL/FP do not want that you would need to learn another API, which kinda makes sense. Because of that, we are currently working on a magic helper to let you use the holy jquery API. 
Something like this:
```
editor.withDomMutations(() => {
    // jquery code comes here and the dom mutations that it triggers are replicated in our VDOM.
});
```

We will need to find a way to incorporate the preview feature in this, but the duplicated code (preview/not preview) won't stay.

Reply:
Yes, NBY explained a bit, seems cool for the developer/design :+1: but indeed, it is probably not straightforward to implement :fearful: